command/agent/consul/syncer.go

```diff
@@ -684,6 +687,20 @@ func (c *Syncer) registerCheck(chkReg *consul.AgentCheckRegistration) error {
 	return c.client.Agent().CheckRegister(chkReg)
 }
 
+// ensureCheckRunning verifies that registerd check is in running state
+func (c *Syncer) ensureCheckRunning(chk *consul.AgentCheckRegistration) {
+	c.registryLock.RLock()
+	defer c.registryLock.RUnlock()
+	cr, ok := c.checkRunners[consulCheckID(chk.ID)]
```

Comment:
I think we can simplify this -

```if cr, ok := c.checkRunners[consulCheckID(chk.ID)]; ok && !cr.Started() {
 		c.logger.Printf("[DEBUG] consul.syncer: starting runner for existing check. %v", chk.ID)
 		cr.Start()
 	}
```

Note that I have also suggested changing the log message slightly.